src/arithmetic/arithmetic_expression.c

```diff
@@ -232,7 +232,32 @@ AR_ExpNode *AR_EXP_FromExpression(RecordMap *record_map, const cypher_astnode_t
 		const cypher_astnode_t *rhs_node = cypher_ast_binary_operator_get_argument2(expr);
 		op->op.children[1] = AR_EXP_FromExpression(record_map, rhs_node);
 		return op;
+	} else if(type == CYPHER_AST_CASE) {
+		/* Determin number of children expressions:
+		 * arg_count = value + 2*alternatives + default. */
+		unsigned int alternatives = cypher_ast_case_nalternatives(expr);
```

Comment:
Does libcypher parser hold different ast nodes or different number of expressions when in general case form?

Reply:
in the simple case we've got:
1. value
2. a variable number of Alternatives (pairs)
3. Default value which might be null

In the generic form value is omitted.

Reply:
so as far as I understand, In the generic form calling `const cypher_astnode_t *expression = cypher_ast_case_get_expression(expr);` will yield NULL? how does the AR_CASEWHEN function can differ between an omitted value (generic) to a null evaluated expression?

Reply:
by inspecting argc,
odd for generic, even for simple

Reply:
oh, missed this update :)